src/main/java/org/fcrepo/camel/FcrepoClient.java

```diff
@@ -81,7 +82,7 @@ public FcrepoResponse head(final URI url)
         final HttpRequestBase request = HttpMethods.HEAD.createRequest(url);
         final HttpResponse response = executeRequest(request);
         final int status = response.getStatusLine().getStatusCode();
-        final String contentType = getContentTypeHeader(response);
+        final Optional<String> maybeContentType = getContentTypeHeader(response);
```

Comment:
This (and all that follows on it below with the use of `Optional::ifPresent`) isn't unarguably great. See [Stuart Marks' explanation](http://stackoverflow.com/questions/23454952/uses-for-java8-optional/23464794#23464794). Long story short: you're shortening code but trading an explicit `null`check for an object creation and an implicit `null`check. I don't have a problem with it, but I know some people wouldn't be enthused.